src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -115,25 +115,29 @@ public abstract class XmlSerializerImplementation
         public virtual XmlSerializer GetSerializer(Type type) { throw new NotSupportedException(); }
     }
 
+    // This enum is intentionally kept outside of the XmlSerializer class since if it would be a subclass
+    // of XmlSerializer, then any access to this enum would be treated by AOT compilers as access to the XmlSerializer
+    // as well, which has a large static ctor which brings in a lot of code. So keeping the enum separate
+    // makes sure that using just the enum itself doesn't bring in the whole of serialization code base.
+#if FEATURE_SERIALIZATION_UAPAOT
+    public enum SerializationMode
```

Comment:
are there any compat concerns regarding moving this type outside?

Reply:
ahh this is only public for UAPAOT

Reply:
As you already noted that is basically a private test hook. We had to make it public for UAPAOT to work correctly, but it should only be used by tests.